acme/client.go

```diff
@@ -42,10 +42,10 @@ type User interface {
 
 // Interface for all challenge solvers to implement.
 type solver interface {
-	Solve(challenge challenge, domain string) error
+	Solve(challenge IdChallenge, domain string) error
```

Comment:
"ID" should be capitalized, as it is an initialism: https://github.com/golang/lint/blob/39d15d55e9777df34cdffde4f406ab27fd2e60c0/lint.go#L706 - otherwise lint is going to throw warnings.